### PR TITLE
Reintroduce .unitypackage building into our build system

### PR DESCRIPTION
**Pull request by wiwei**
Build.ps1 has been out of commission for a while now, and building .unitypackages has been a manual step for us prior to release.

This change removes build.ps1 and moves some of its functionality into a script that is just designed to do .unitypackages, and in a way that is also compatible with the way the rest of our pipelines work now.

This also updates our CI pipeline (not PR pipeline) to generate .unitypackage artifacts in the build (the Foundation and Examples package)

### CODE REVIEW

File: scripts/packaging/unitypackage.ps1 [hunk 1]
```diff
@@ -205,24 +161,34 @@ foreach ($entry in $packages.GetEnumerator()) {
...
+        Remove-Job $ljob
+
+        Stop-Process $proc
 
-        $process = Start-UnityEditor @unityArgs
-        if (($process.ExitCode -eq 0) -and (Test-Path $unityPackagePath)) {
+        if (($proc.ExitCode -eq 0) -and (Test-Path $unityPackagePath)) {
             Write-Verbose "Successfully created $unityPackagePath"
         }
         else { Write-Error "Failed to create $unityPackagePath" }
```
Comment (macborow) [inline scripts/packaging/unitypackage.ps1]: The script should explicitly set the error code ($rc = ... followed by exit $rc at the end of the script).
If unset, the RC will equal to the error code from the last command which doesn't really work if there are some clean up steps involved).

Comment (wiwei) [reply scripts/packaging/unitypackage.ps1]: doh, I missed this update. I'll do what we did in unitybuild.yml

**Comment by MenelvagorMilsom** [comment]
@wiwei Have you tested this on a CI (rather than PR) pipeline?

**Comment by wiwei** [comment]
> @wiwei Have you tested this on a CI (rather than PR) pipeline?

The actual CI test is happening now:

https://dev.azure.com/aipmr/MixedRealityToolkit-Unity-CI/_build/results?buildId=3316

But I had tested this on a pipeline here:

https://dev.azure.com/aipmr/MixedRealityToolkit-Unity-CI/_build/results?buildId=3286

And used the generated YAML to populate this stuff.

**Comment by MenelvagorMilsom** [comment]
Noice

**Comment by wiwei** [comment]
> Noice

Noice

**Pull request merged**